### ISSUE TITLE
[Bug] GPU memory  unreasonable increase

### ISSUE THREAD

**Issue by yyh-xjtu** [open-mmlab/mmrotate#927]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmrotate/issues) and [Discussions](https://github.com/open-mmlab/mmrotate/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmrotate.readthedocs.io/en/1.x/notes/4_faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmrotate) or [latest version (1.x)](https://github.com/open-mmlab/mmrotate/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

1.x branch https://github.com/open-mmlab/mmrotate/tree/1.x

### Environment

sys.platform: win32
Python: 3.8.17 (default, Jul  5 2023, 20:44:21) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: NVIDIA GeForce RTX 4090
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8
NVCC: Cuda compilation tools, release 11.8, V11.8.89
MSVC: 用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.36.32532 版
GCC: n/a
PyTorch: 1.13.1
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX2
  - CUDA Runtime 11.7
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.5
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=C:/cb/pytorch_1000000000000/work/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL
 -openmp:experimental -IC:/cb/pytorch_1000000000000/work/mkl/include -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO, LAPACK_INFO=mkl, PERF_WITH_AVX=1
, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_ROCM=OFF,

TorchVision: 0.14.1
OpenCV: 4.8.0
MMEngine: 0.8.4
MMRotate: 1.0.0rc1+3500994

### Reproduces the problem - code sample

oriented-rcnn-le90_r50_fpn_1x_dota.py with dota_ms.py

### Reproduces the problem - command or script

python ./tools/train.py path_to\oriented-rcnn-le90_r50_fpn_1x_dota_ms.py 

### while training ,The time spent gradually increases，and the gpu memory increased from 14G to 23.8G , and another 28G shared gpu memory was used.  so at epoch 2 ,there are totally 50G memory used.  

09/03 15:43:31 - mmengine - INFO - Epoch(train)  [1][ 2600/34163]  lr: 5.0000e-03  eta: 1 day, 16:19:54  **time: 0.3499**  data_time: 0.0011  memory: 7828  grad_norm: 4.0232  loss: 0.8435  loss_rpn_cls: 0.0686  loss_rpn_bbox: 0.1694  loss_cls: 0.2901  acc: 93.2617  loss_bbox: 0.3153
09/03 20:38:29 - mmengine - INFO - Epoch(train)  [1][32950/34163]  lr: 5.0000e-03  eta: 2 days, 11:11:29  **time: 1.4869**  data_time: 0.0021  memory: 7269  grad_norm: 2.7446  loss: 0.4138  loss_rpn_cls: 0.0278  loss_rpn_bbox: 0.0912  loss_cls: 0.1518  acc: 97.7539  loss_bbox: 0.1429
09/04 07:54:22 - mmengine - INFO - Epoch(train)  [2][ 5600/34163]  lr: 5.0000e-03  eta: 6 days, 9:02:03  **time: 5.9570**  data_time: 0.0009  memory: 6372  grad_norm: 2.7575  loss: 0.3757  loss_rpn_cls: 0.0244  loss_rpn_bbox: 0.0688  loss_cls: 0.1548  acc: 93.9453  loss_bbox: 0.1278

### Additional information

dataset : dota trainval with ms (1024,1024,500) (0.5,1.0.1.5)

![image](https://github.com/open-mmlab/mmrotate/assets/70417599/0b45c37a-fa44-49fb-af85-7e0c8b3e1d8e)
